bigframes/core/eval.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import dataclasses
+from typing import Optional
+
+import pandas
+
+import bigframes.dataframe as dataframe
+import bigframes.dtypes
+import bigframes.series as series
+import third_party.bigframes_vendored.pandas.core.computation.parsing as vendored_pandas_eval_parsing
+
+
+def eval(df: dataframe.DataFrame, expr: str, target: Optional[dataframe.DataFrame]):
```

Comment:
Docstring? I'm especially interested in what the `target` argument does.

Reply:
added